js-envs/src/main/scala/org/scalajs/jsenv/nodejs/JSDOMNodeJSEnv.scala

```diff
@@ -0,0 +1,110 @@
+/*                     __                                               *\
+**     ________ ___   / /  ___      __ ____  Scala.js sbt plugin        **
+**    / __/ __// _ | / /  / _ | __ / // __/  (c) 2013, LAMP/EPFL        **
+**  __\ \/ /__/ __ |/ /__/ __ |/_// /_\ \    http://scala-js.org/       **
+** /____/\___/_/ |_/____/_/ | |__/ /____/                               **
+**                          |/____/                                     **
+\*                                                                      */
+
+
+package org.scalajs.jsenv.nodejs
+
+import java.io.{Console => _, _}
+
+import org.scalajs.core.tools.io._
+import org.scalajs.core.tools.jsdep.ResolvedJSDependency
+import org.scalajs.jsenv._
+
+class JSDOMNodeJSEnv(
+  nodejsPath: String = "node",
+  addArgs: Seq[String] = Seq.empty,
+  addEnv: Map[String, String] = Map.empty
+) extends AbstractNodeJSEnv(nodejsPath, addArgs, addEnv, sourceMap = false) {
+
+  protected def vmName: String = "Node.js with JSDOM"
+
+  override def jsRunner(libs: Seq[ResolvedJSDependency],
+      code: VirtualJSFile): JSRunner = {
+    new DOMNodeRunner(libs, code)
+  }
+
+  override def asyncRunner(libs: Seq[ResolvedJSDependency],
+      code: VirtualJSFile): AsyncJSRunner = {
+    new AsyncDOMNodeRunner(libs, code)
+  }
+
+  override def comRunner(libs: Seq[ResolvedJSDependency],
+      code: VirtualJSFile): ComJSRunner = {
+    new ComDOMNodeRunner(libs, code)
+  }
+
+  protected class DOMNodeRunner(libs: Seq[ResolvedJSDependency], code: VirtualJSFile)
+      extends ExtRunner(libs, code) with AbstractDOMNodeRunner
+
+  protected class AsyncDOMNodeRunner(libs: Seq[ResolvedJSDependency], code: VirtualJSFile)
+      extends AsyncExtRunner(libs, code) with AbstractDOMNodeRunner
+
+  protected class ComDOMNodeRunner(libs: Seq[ResolvedJSDependency], code: VirtualJSFile)
+      extends AsyncDOMNodeRunner(libs, code) with NodeComJSRunner {
+    override protected def initFiles(): Seq[VirtualJSFile] =
+      super.initFiles ++ comSetup
+  }
+
+  protected trait AbstractDOMNodeRunner extends AbstractNodeRunner {
+
+    protected def codeWithJSDOMContext(): Seq[VirtualJSFile] = {
+      val scriptsJSPaths = getLibJSFiles().map {
+        case file: FileVirtualFile => file.path
+        case file                  => libCache.materialize(file).getAbsolutePath
+      }
+      val scriptsStringPath = scriptsJSPaths.map('"' + _ + '"')
```

Comment:
That is too weak. File names can contain weird characters. We have a function somewhere to properly escape JS strings (it's called `escapeJS`, I think).